### ISSUE TITLE
vec( $str, -1, 1) dumps core

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2710]
Migrated from [rt.perl.org#4432](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=4432) (status was 'resolved')

Searchable as RT4432$

**Comment by p5pRT** [comment]
### From anno4000@lublin.zrz.tu-berlin.de

#### Created by anno4000@lublin.zrz.tu-berlin.de



Accessing vec\( $str\, \-1\, 1\) dumps core after an attempt to assign a
non\-zero value to vec\( $str\, \-1\, 1\)\.  The assignment itself seems to
succeed\.  The program below illustrats this behavior\.

Anno

my $vec = '';
print "one&#8203;: "\, vec\( $vec\, \-1\, 1\)\, "\\n";   \# reads 0
vec\( $vec\, \-1\, 1\) = 1;                    \# assigning seems to work
print "two\\n";                            \# \.\.\.as shown here
print "three&#8203;: "\, vec\( $vec\, \-1\, 1\)\, "\\n"; \# but this segfaults

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.6.0:

Configured by anno4000 at Wed Jun  7 10:42:24 MEST 2000.

Summary of my perl5 (revision 5.0 version 6 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.0.36, archname=i586-linux
    uname='linux lublin.zrz.tu-berlin.de 2.0.36 #5 fre apr 23 14:43:16 cest 1999 i586 unknown '
    config_args='-de'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cc', optimize='-O2', gccversion=2.7.2.3
    cppflags='-I/usr/local/include'
    ccflags ='-I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldb -ldl -lm -lc -lposix -lcrypt
    libc=/lib/libc-2.0.7.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.0:
    /usr/local/lib/perl5/5.6.0/i586-linux
    /usr/local/lib/perl5/5.6.0
    /usr/local/lib/perl5/site_perl/5.6.0/i586-linux
    /usr/local/lib/perl5/site_perl/5.6.0
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.6.0:
    HOME=/opt/home/anno4000
    LANG=de
    LANGUAGE (unset)
    LC_ALL=de_DE
    LD_LIBRARY_PATH=/usr/X11R6/lib:/usr/local/LessTif/Motif1.2/lib
    LOGDIR (unset)
    PATH=/opt/home/anno4000/bin:/sbin:/usr/local/bin:/bin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/usr/openwin/bin:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/tcsh


```
</details>

**Comment by p5pRT** [comment]
### From @simoncozens

On Fri\, Oct 13\, 2000 at 03&#8203;:18&#8203;:01PM \+0000\, Anno Siegel wrote&#8203;:
> Accessing vec\( $str\, \-1\, 1\) dumps core after an attempt to assign a
> non\-zero value to vec\( $str\, \-1\, 1\)\.  The assignment itself seems to
> succeed\.  The program below illustrats this behavior\.

Seems fixed in bleadperl&#8203;:

one&#8203;: 0
Assigning to negative offset in vec at \- line 3\.

**Comment by p5pRT** [comment]
### From @simoncozens

On Fri\, Oct 13\, 2000 at 04&#8203;:23&#8203;:55PM \+0100\, Simon Cozens wrote&#8203;:
> On Fri\, Oct 13\, 2000 at 03&#8203;:18&#8203;:01PM \+0000\, Anno Siegel wrote&#8203;:
> > Accessing vec\( $str\, \-1\, 1\) dumps core after an attempt to assign a
> > non\-zero value to vec\( $str\, \-1\, 1\)\.  The assignment itself seems to
> > succeed\.  The program below illustrats this behavior\.
> 
> Seems fixed in bleadperl&#8203;:

In fact\, this is bug 20000821\.008; can some friendly admin close this?

**Comment by p5pRT** [comment]
### From @jhi

On Fri\, Oct 13\, 2000 at 04&#8203;:26&#8203;:28PM \+0100\, Simon Cozens wrote&#8203;:
> On Fri\, Oct 13\, 2000 at 04&#8203;:23&#8203;:55PM \+0100\, Simon Cozens wrote&#8203;:
> > On Fri\, Oct 13\, 2000 at 03&#8203;:18&#8203;:01PM \+0000\, Anno Siegel wrote&#8203;:
> > > Accessing vec\( $str\, \-1\, 1\) dumps core after an attempt to assign a
> > > non\-zero value to vec\( $str\, \-1\, 1\)\.  The assignment itself seems to
> > > succeed\.  The program below illustrats this behavior\.
> > 
> > Seems fixed in bleadperl&#8203;:
> 
> In fact\, this is bug 20000821\.008; can some friendly admin close this?

\<ka\-ching>The sound of a bug closing\.\</ka\-ching>
